src/critical_section.rs

```diff
@@ -0,0 +1,44 @@
+//! Single-core critical section implementation using the [`critical_section`]
+//! crate. Only the [`with`] function is publicly exposed.
+
+#[cfg(all(target_arch = "msp430", feature = "critical-section-single-core"))]
+mod critical_section_single_core {
+    use crate::{interrupt, register};
+    use critical_section::RawRestoreState;
+
+    struct CriticalSection;
+    critical_section::set_impl!(CriticalSection);
+
+    unsafe impl critical_section::Impl for CriticalSection {
+        // Without #[inline(never)] attribute, rustc tends to misoptimize for
+        // size on functions that use interrupt::free() or 
+        // critical_section::with().
+        // I don't believe #[inline] or #[inline(always)] hints work for
+        // functions marked as "extern" in another crate (which is the case
+        // for the acquire()/release() here- see critical_section crate).
+        #[cfg_attr(any(feature = "outline-cs", feature = "outline-cs-acq"), inline(never))]
```

Comment:
You don't need to check for `outline-cs` here, because `outline-cs` already implies `outline-cs-acq` and `outline-cs-rel`, according to Cargo.toml.